src/main/java/Duke.java

```diff
@@ -1,10 +1,126 @@
+     * Replace this stub with your completed method.
+     */
+    String getResponse(String input) {
+        return "Duke heard: " + input;
+    }
+
+    /**
+     * The constructor for the Duke class.
+     */
+    public Duke() {
+        ui = new Ui();
+        tasks = new TaskList();
+        storage = new Storage();
+        parser = new Parser();
+    }
+
+    /**
+     * The main method for Duke.
+     * @param args For the main method.
+     * @throws DukeException Throws DukeException.
+     * @throws IOException Throws IOException.
+     */
+    public static void main(String[] args) throws DukeException, IOException {
+        new Duke().run();
+    }
+
+
+    /**
+     * To run the duke program.
+     * @throws DukeException Throws DukeException.
+     * @throws IOException Throws IOException.
+     */
+    public void run() throws DukeException, IOException {
+        storage.loadData();
+        ui.welcomeMessage();
+        ui.reply("    What can I do for you");
+
+        Scanner sc = new Scanner(System.in);
+        String command;
+        while (!(command = sc.nextLine()).equals("bye")) {
+            try {
+                handle(command);
+            }
+            catch (DukeException ex) {
+                ui.showError(ex.getMessage());
+            }
+
+        }
+
+        storage.saveData();
+        ui.reply("    Bye. Hope to see you again soon!");
+    }
+
+    /**
+     * Handles the command provided by the user.
+     * @param command The user command.
+     * @throws DukeException Throws DukeException.
+     */
+    public void handle(String command) throws DukeException{
+        if (command.equals("list")) {
+            ui.printList();
+        } else {
+            String type = parser.getType(command);
+            if (type.equals("done") || type.equals("delete")) {
+                int taskNo = parser.getTaskIndex(command);
+                if (taskNo > tasks.numOfTasks() || taskNo <= 0) {
+                    throw new DukeException("☹ OOPS!! Not a valid number");
+                } else {
+                    if (type.equals("done")) {
+                        tasks.getTask(taskNo - 1).markAsDone();
+                        ui.doneMessage(tasks.getTask(taskNo - 1));
+                    } else {
+                        ui.deleteMessage(tasks.getTask(taskNo - 1));
+                        tasks.removeTask(taskNo - 1);
+                        Task.totalTasks--;
+                    }
+                }
+            } else if (type.equals("find")) {
+                ArrayList<Integer> taskFound = new ArrayList<>();
+                for (int i = 0; i < tasks.numOfTasks(); i++) {
+                    if (tasks.getTask(i).description.contains(command.split(" ")[1])) {
+                        taskFound.add(i);
+                    }
+                }
+                ui.printSelected(taskFound);
+            } else {
+                // create new task -> add to tasks -> reply
```

Comment:
Comments are written to help the readers to understand the code. So better you write useful and descriptive comments